src/Mutator/Definition.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator;
+
+use Webmozart\Assert\Assert;
+
+final class Definition
+{
+    private $description;
+    private $category;
+    private $remedies;
+
+    public function __construct(
+        string $description,
+        string $category,
+        ?string $remedies
```

Comment:
From the tests: `'This text is for providing guidelines on how to kill the mutant.'`

It would be more clear to either rename this parameter or add this comment to a property.

Reply:
Worth mentioning that there is little mutators where I started to document it. The reason being that for most of them, you can provide "default" guidelines based on the mutator category. For example:

```
Semantic reductions exposes unused semantics. If the semantics are unneeded, they should be removed. Otherwise they should be tested.
```

So I only documented it when I deemed worthy of it more as an internal documentation for now. It's definitely something that will need some adjustments (e.g. providing a default guideline instead of nothing) when we decide to expose it to the consumer via the doc for example. In other words: I laid down the foundation for it and attempt to better document the tricky cases I identified, but it will definitely require more work before exposing it to the users